### ISSUE TITLE
Create `raw` bindings to cspice

### ISSUE THREAD

**Issue by ChristopherRabotin** [ChristopherRabotin/SPICE#1]
- [x] Allow direct unfettered access to `cspice` from Rust which might have to wrapped with the `unsafe` keyword.
- [x] Add a test suite similar to that of [SpiceyPy](https://github.com/AndrewAnnex/SpiceyPy)

**Comment by ChristopherRabotin** [comment]
From current code, add error fetching and a `Result<,>` returning macro for calls to raw. Maybe something like the following:
```rust
macro_rules! raw {
    (($instr:expr)+) => {{
        unsafe{
             ::raw::$instr;
             if ::raw::failed_c() == 1 {
                 return Err();
             }
             return Ok();
        }
    }};
}
```

**Comment by ChristopherRabotin** [comment]
Closed in #4
